source/tests/system/nirfmxwlan_restricted_driver_api_tests.cpp

```diff
@@ -0,0 +1,153 @@
+  EXPECT_RESPONSE_SUCCESS(client::cfg_iq_power_edge_trigger(wlan_stub, session, "", "0", IQ_POWER_EDGE_TRIGGER_SLOPE_RISING, -20.0, 0.0, TRIGGER_MINIMUM_QUIET_TIME_MODE_AUTO, 5e-6, IQ_POWER_EDGE_TRIGGER_LEVEL_TYPE_RELATIVE, true));
+  EXPECT_RESPONSE_SUCCESS(client::cfg_standard(wlan_stub, session, "", STANDARD_802_11_AG));
+  EXPECT_RESPONSE_SUCCESS(client::cfg_channel_bandwidth(wlan_stub, session, "", 20e06));
+  EXPECT_RESPONSE_SUCCESS(client::select_measurements(wlan_stub, session, "", MEASUREMENT_TYPES_OFDMMODACC, true));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_measurement_length(wlan_stub, session, "", 0, 16));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_frequency_error_estimation_method(wlan_stub, session, "", OFDMMODACC_FREQUENCY_ERROR_ESTIMATION_METHOD_PREAMBLE_AND_PILOTS));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_amplitude_tracking_enabled(wlan_stub, session, "", OFDMMODACC_AMPLITUDE_TRACKING_ENABLED_FALSE));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_phase_tracking_enabled(wlan_stub, session, "", OFDMMODACC_PHASE_TRACKING_ENABLED_TRUE));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_symbol_clock_error_correction_enabled(wlan_stub, session, "", OFDMMODACC_SYMBOL_CLOCK_ERROR_CORRECTION_ENABLED_TRUE));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_channel_estimation_type(wlan_stub, session, "", OFDMMODACC_CHANNEL_ESTIMATION_TYPE_REFERENCE));
+  EXPECT_RESPONSE_SUCCESS(client::ofdm_mod_acc_cfg_averaging(wlan_stub, session, "", OFDMMODACC_AVERAGING_ENABLED_FALSE, 10));
+}
+
+nidevice_grpc::Session init_session(const client::StubPtr& stub, const std::string& model)
+{
+  return init_session(stub, model, "FakeDevice");
+}
+
+TEST_F(NiRFmxWLANRestrictedDriverApiTests, GetChannelListBehavior_ReturnsNonEmptyResult)
+{
+  const auto session = init_session(stub(), PXI_5841);
+  EXPECT_SUCCESS(session, client::select_measurements(stub(), session, "", MeasurementTypes::MEASUREMENT_TYPES_OFDMMODACC, true));
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto response = restricted_client::get_channel_list(restricted_stub(), session, "", 0);
+  EXPECT_SUCCESS(session, response);
+  EXPECT_THAT(response.center_frequencies(), Not(IsEmpty()));
+  EXPECT_THAT(response.channel_bandwidths(), Not(IsEmpty()));
+  EXPECT_GT(response.actual_array_size(), 0);
+}
+
+TEST_F(NiRFmxWLANRestrictedDriverApiTests, OFDMModAccFromExample_FetchCommonPilotErrorTradeIndB_ReturnsNonEmptyResult)
+{
+  char* resource_name = "RFSA";
+  auto instr_stub = create_stub<nirfmxinstr_grpc::NiRFmxInstr>();
+  const auto session = init_session(stub(), PXI_5841);
+  configure_OFDMModAccFromExample(instr_stub, session, stub());
+  EXPECT_SUCCESS(session, client::initiate(stub(), session, "", ""));
+
+  const auto response = restricted_client::ofdm_mod_acc_fetch_common_pilot_error_trace_ind_b(restricted_stub(), session, "", 10.0);
+  EXPECT_SUCCESS(session, response);
```

Comment:
Should we be asserting on any of the other response values no this one?

Reply:
Added checks for other response values.